generated/nidcpower/nidcpower/lcr_measurement.py

```diff
@@ -0,0 +1,210 @@
+        ("d"                 , "Dissipation factor"                                                  ),  # noqa: E202,E203
+        ("measurement_mode"  , "Measurement mode"                                                    ),  # noqa: E202,E203
+        ("dc_in_compliance"  , "DC in compliance"                                                    ),  # noqa: E202,E203
+        ("ac_in_compliance"  , "AC in compliance"                                                    ),  # noqa: E202,E203
+        ("unbalanced"        , "Unbalanced"                                                          ),  # noqa: E202,E203
+    ]
+
+    def __init__(self, data):
+        """LCRMeasurement
+
+        Creates and returns an instance of LCRMeasurement.
+
+        Args:
+            data (struct_NILCRMeasurement): The LCR measurement ctypes instance returned by the driver.
+        """
+        self._data = data
+        self.channel = ""
+        self.vdc = data.vdc
+        self.idc = data.idc
+        self.stimulus_frequency = data.stimulus_frequency
+        self.ac_voltage = complex(data.ac_voltage_real, data.ac_voltage_imaginary)
+        self.ac_current = complex(data.ac_current_real, data.ac_current_imaginary)
+        self.z = complex(data.z_real, data.z_imaginary)
+        self.series_lcr = LCRMeasurement.LCR(
+            inductance=data.ls, capacitance=data.cs, resistance=data.rs
+        )
+        self.parallel_lcr = LCRMeasurement.LCR(
+            inductance=data.lp, capacitance=data.cp, resistance=data.rp
+        )
+        self.d = data.d
+        self.measurement_mode = enums.InstrumentMode(data.measurement_mode)
+        self.dc_in_compliance = bool(data.dc_in_compliance)
+        self.ac_in_compliance = bool(data.ac_in_compliance)
+        self.unbalanced = bool(data.unbalanced)
+
+    @property
+    def z_magnitude_and_phase(self):
+        """z_magnitude_and_phase
+
+        Get the **magnitude** (float), in ohms and **phase angle** (float), in degrees of the
+        complex impedance.
+        """
+        return self._data.z_magnitude, self._data.z_phase
+
+    @property
+    def y(self):
+        """y
+
+        Get the complex admittance (complex).
+        """
+        return complex(self._data.y_real, self._data.y_imaginary)
+
+    @property
+    def y_magnitude_and_phase(self):
+        """y_magnitude_and_phase
+
+        Get the **magnitude** (float), in siemens and **phase angle** (float), in degrees of the
+        complex admittance.
+        """
+        return self._data.y_magnitude, self._data.y_phase
+
+    @property
+    def q(self):
+        """q
+
+        Get the quality factor (float) of the circuit. The dimensionless quality factor is inversely
+        proportional to the degree of damping in a system. Q is the reciprocal of D, the dissipation
+        factor.
+        """
+        return self._data.q
```

Comment:
Why are some properties implemented this way (`@property` decorator) and others by setting an internal member during `__init__()` (like `self.ac_voltage`)?

The inconsistency is cofusing if needless.

Reply:
Tracked by #1781.